zegami_sdk/collection.py

```diff
@@ -291,6 +305,104 @@ def imageset_index_to_row_index(self, imageset_index, source=0) -> int:
             raise IndexError(
                 "Invalid imageset index {} for this source"
                 .format(imageset_index))
+    
+    def add_snapshot(self, name, desc, snapshot):
+        url = '{}/{}/project/{}/snapshots/{}/snapshots'.format(
+            self.client.HOME, self.client.API_0, self.workspace_id, self.id)
+        payload = {
+            'name': name,
+            'description': desc,
+            'snapshot': snapshot,
+            'version': 3,
+        }
+        r = self.client._auth_post(url, json.dumps(payload), return_response=True)
+        return r
+
+    def add_feature_pipeline(self, name, steps, source=0, generate_snapshot=False):
+        # get the source
+        source = self._parse_source(source)
+        node_group = [
+            'source_{}'.format(source.name),
+            'collection_{}'.format(self.id),
+        ]
+
+        mRMR_params = steps[0]['params']
+
+        # find the feature extraction node
+        source_feature_extraction_node = self.get_feature_extraction_imageset_id(source)
+
+        join_dataset_id = source._imageset_dataset_join_id
+        imageset_parents = [source_feature_extraction_node]
+        dataset_parents = [self._dataset_id, join_dataset_id]
+
+        mRMR_node = add_node(
+            self.client,
+            self.workspace,
+            action=steps[0]['action'],
+            params=mRMR_params,
+            type="imageset",
+            dataset_parents=dataset_parents,
+            imageset_parents=imageset_parents,
+            processing_category='image_clustering',
+            node_group=node_group,
+            name="mRMR imageset for {} of {}".format(source.name, self.name),
+        )
+
+        cluster_params = steps[1]['params']
+        cluster_params["out_column_titles"] = [
+            "mRMR Image Similarity x ({})".format(source.name),
+            "mRMR Image Similarity y ({})".format(source.name),
+        ]
+        target_column_stripped = mRMR_params['target_column'].replace(' ', '').replace('-', '').lower()
+        cluster_params["out_columns"] = [
+            "mRMR_image_similarity_x_{}_{}_{}".format(source.name, target_column_stripped, mRMR_params['K']),
+            "mRMR_image_similarity_y_{}_{}_{}".format(source.name, target_column_stripped, mRMR_params['K']),
+        ]
```

Comment:
Because we allow the user to set `n_components`, we should use `out_column_title_prefix`, `out_column_name_prefix` and `out_column_start_order` here.
Again, the titles should be based on the name the user has passed in.